### PR TITLE
Android: Numerous Material Design additions to the UI.

### PR DESCRIPTION
**Pull request by sigmabeta**
- Games are now sorted on the main screen into tabs by platform (GC / Wii / Wiiware).
- Toolbar collapses if the user scrolls down to fit more games on screen.
- Slight refactoring of some files.
- Uses AppCompat + Material Design Support libraries to not have to hack together slightly incorrect versions of MD elements.

This might break stuff on TV devices (I haven't tried yet) but once this is merged I'd like to begin work on the TV-specific UI.

Here's what it looks like:

device-2015-06-21-193527

I don't have a Wiiware graphic at all, and the GC/Wii icons are very low resolution, so once I have better ones I'll be sure to throw those in.

### CODE REVIEW

**Comment by sigmabeta** [comment]
Here's what it looks like:

device-2015-06-21-193527


I don't have a Wiiware graphic at all, and the GC/Wii icons are very low resolution, so once I have better ones I'll be sure to throw those in.

**Comment by sigmabeta** [comment]
Paging @Parlane to run necessary updates on buildbot.

If my CLI-ese is correct, the necessary command should be:

`./android update sdk --no-ui --filter extra`

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java [hunk 1]
```diff
@@ -9,71 +8,79 @@
...
 {
 	private static final int REQUEST_ADD_DIRECTORY = 1;
 
-	private static final int LOADER_ID_GAMES = 1;
-	// TODO When each platform has its own tab, there should be a LOADER_ID for each platform.
+	/**
+	 * It is important to keep track of loader ID separately from platform ID (see Game.java)
+	 * because we could potentially have Loaders that load things other than Games.
+	 */
+	public static final int LOADER_ID_ALL = 100; // TODO
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java]: This TODO should probably be elaborated or removed.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java [hunk 2]
```diff
@@ -205,25 +235,72 @@ public void onLoadFinished(Loader<Cursor> loader, Cursor data)
...
+			// TODO case LOADER_ID_ALL:
+
 			default:
 				Log.e("DolphinEmu", "Bad ID passed in.");
 		}
 
+		if (fragment != null)
+		{
+			fragment.onLoadFinished(loader, data);
+
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java]: Unnecessary newline

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java [hunk 3]
```diff
@@ -205,25 +235,72 @@ public void onLoadFinished(Loader<Cursor> loader, Cursor data)
...
+				break;
+
+			case LOADER_ID_WIIWARE:
+				fragment = getPlatformFragment(Game.PLATFORM_WII_WARE);
+				break;
+
+			// TODO case LOADER_ID_ALL:
+
+			default:
+				Log.e("DolphinEmu", "Bad ID passed in.");
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java]: missing break.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java [hunk 4]
```diff
@@ -205,25 +235,72 @@ public void onLoadFinished(Loader<Cursor> loader, Cursor data)
...
+				break;
+
+			case LOADER_ID_WIIWARE:
+				fragment = getPlatformFragment(Game.PLATFORM_WII_WARE);
 				break;
 
+			// TODO case LOADER_ID_ALL:
+
 			default:
 				Log.e("DolphinEmu", "Bad ID passed in.");
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java]: Should probably add a break here.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/PlatformPagerAdapter.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import android.text.style.ImageSpan;
+
+import org.dolphinemu.dolphinemu.R;
+import org.dolphinemu.dolphinemu.fragments.PlatformGamesFragment;
+
+public class PlatformPagerAdapter extends FragmentPagerAdapter
+{
+	private Context mContext;
+
+	private final static int TAB_ICONS[] = new int[]{
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/PlatformPagerAdapter.java]: `new int[]` isn't necessary.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java (hunk 1 above)
Comment (sigmabeta) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java]: It's just a reminder that LOADER_ID_ALL doesn't actually do anything and will need to be implemented.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java (hunk 4 above)
Comment (sigmabeta) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java]: A break in the default case? Wouldn't that be unnecessarily verbose? It wouldn't any any functional effect.

Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/MainActivity.java]: right, however static analysis tools will flag that.

**Comment by sigmabeta** [comment]
@lioncash I've implemented your requested fixes, even if it doesn't look like it above (some of the fixes came on the next line)

**Pull request merged**